File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/2-1413 Minimum Value to Get Positive Step by Step Sum/main.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int min_val = 0;
        int total = 0;
        for (int num : nums) {
            total += num;
            min_val = min(min_val, total);
        }

        return -min_val + 1;
    }
};

File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/2-1413 Minimum Value to Get Positive Step by Step Sum/main.ts
function minStartValue(nums: number[]): number {
  let minVal = 0;
  let total = 0;
  for (let i = 0; i < nums.length; i++) {
    total += nums[i];
    minVal = Math.min(minVal, total);
  }

  return -minVal + 1;
}
